Exercise-02/Exercise02/main.swift

```
@@ -7,63 +7,386 @@
+    
+    let (city, cityRecords) = getCityRecords(records: records)
+    let hoursInt: Int = readHour()
+    let dayInt: Int = readDay()
+    // Checking if cityRecords contain the hour and day
+    let tempRecords = cityRecords.filter { $0.day != dayInt || $0.hour != hoursInt }
+    guard tempRecords.count != cityRecords.count else {
+        print("Record not found")
+        return
+    }
+    records[city] = tempRecords
+    print("Success!")
+}
+
+/// Prints the records in increasing or decreasing order
+/// - Parameters:
+///   - records: the record of the temperatures
+///   - unit: unit in which the temperatures need to be displayed
+func printRecords (records: TempManual, unit: String) {
+    
+    let (city, cityRecords) = getCityRecords(records: records)
+    var order: String = ""
+    while true {
+        print("Please enter the order(i for increasing, d for decreasing)")
+        if let orderTemp = readLine(), orderTemp == "i" || orderTemp == "d" {
+            order = orderTemp
+            break
+        }
+        print("Invalid order entered")
+    }
+    let dayInt: Int = readDay()
+    print("For the city \(city) the records for the day \(dayInt) are: ")
+    switch order {
+    case "i":
+        for (temperature, hour, day) in cityRecords {
+            guard let temperatureInUnit = temperatureToUnit(temperature: temperature, unit: unit) else {
+                return
+            }
+            if day == dayInt {
+                print("temperature: \(temperatureInUnit) day: \(day) hour: \(hour)")
+            }
+        }
+    case "d":
+        for (temperature, hour, day) in cityRecords.reversed() {
+            guard let temperatureInUnit = temperatureToUnit(temperature: temperature, unit: unit) else {
+                return
+            }
+            if day == dayInt {
+                print("temperature: \(temperatureInUnit) \(unit) day: \(day) hour: \(hour)")
+            }
+        }
+    default:
+        return
```

Again, here I would probably handle it more explicitly, instead of failing silently